src/actions/write/evolveState.ts

```diff
@@ -16,31 +12,16 @@ export const evolveState = async (
     throw new ContractError(NON_CONTRACT_OWNER_MESSAGE);
   }
 
-  // update the auction settings object
-  state.settings.auctions = AUCTION_SETTINGS;
+  const totalBalances = Object.values(state.balances).reduce(
+    (total, current) => total + current,
+    0,
+  );
 
-  // update existing auctions to use the new settings
-  for (const auction of Object.keys(state.auctions)) {
-    state.auctions[auction] = {
-      ...state.auctions[auction],
-      settings: AUCTION_SETTINGS,
-    };
-  }
-
-  // TODO: Should this be using previous contracts DF values?
-  // update demand factoring
-  state.demandFactoring = {
-    periodZeroBlockHeight: +SmartWeave.block.height,
-    currentPeriod: 0,
-    trailingPeriodPurchases: [0, 0, 0, 0, 0, 0, 0],
-    trailingPeriodRevenues: [0, 0, 0, 0, 0, 0, 0],
-    purchasesThisPeriod: 0,
-    revenueThisPeriod: 0,
-    demandFactor: DEMAND_FACTORING_SETTINGS.demandFactorBaseValue,
-    consecutivePeriodsWithMinDemandFactor: 0,
-  };
+  const diff = TOTAL_IO_SUPPLY - totalBalances;
```

Comment:
Is this forgetting to check for IO lodged in gateway stakes and auctions?

Reply:
yep! will come back and add

Reply:
[9ab4ce7](https://github.com/ar-io/testnet-contract/pull/130/commits/9ab4ce738c14305d420cd18ed88572ad1443655e)